src/views/components/Loader.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { View, StyleSheet, useWindowDimensions, ActivityIndicator, Text } from 'react-native';
+import COLORS from '../../const/colors';
+
+const Loader = ({ visible = false }) => {
+  const { height, width } = useWindowDimensions();
+  return visible && <View style={[styles.container, {height, width}]}>
```

Comment:
Component should return something, if you don't need to display anything, then you should return null. In this case, it will throw an error. If it's not visible, then it will return undefined.